keps/sig-release/1731-publishing-packages/README.md

```diff
@@ -149,51 +183,236 @@ Scenario: [...]
+- Binary published by our release process would differ to binaries built by OBS
+  - Additional efforts would be needed to get reproducible builds working
+  - We would also lose cosign signatures for binaries built by OBS
+
+`kubepkg` will be extended with a subcommand to create a tarball with all required binaries and files (e.g. systemd
+units and config files). The tarball is supposed to be created with the maximum compression to save on bandwidth and
+storage. The structure of the tarball is supposed to be:
+
+- Root of tarball:
+  - LICENSE file
+  - README.md file
+  - All accompanying files (e.g. systemd units)
+  - Subdirectory for each target architecture:
+    - Binary for that architecture (e.g. `kubectl`)
+
+#### Package Specs
+
+There are two key changes to the package specs compared to what specs we have at the time of writing this KEP:
+
+- We'll maintain specs only for the RPM-based distros
+- We'll have a dedicated spec for each package
+  - Right now, for RPM-based distros, we have one spec file that builds all packages
+  - This is to make it easier to maintain and update those spec files / packages, as well as, to make it easier for
+    distributors to consume and use those spec files
+
+The starting point for creating RPM specs is going to be the [RPM specs currently embedded in `kubepkg`][kubepkg-rpm].
+The following changes are needed to those RPM specs:
+
+- Parametrize specs so the build tooling is able to pick a binary for the correct target architecture
+- Provide additional metadata needed for building Debian-based packages
+- Ensure all spec files are passing rpm-lint
+
+The reason for dropping deb specs is that maintaining and generating those specs is more complicated than
+maintaining RPM specs. Considering that we use pre-built binaries, we can easily convert RPM specs to Debian specs
+automatically using the [`debbuild` tool][debbuild]. The `debbuild` tool is already available in the OBS pipeline.
+This tool can also be used by distributors if they want to build deb packages on their own.
+
+The RPM specs will be generated by `kubepkg`, which already supports this. That said, we only need to update the spec
+files.
+
+[kubepkg-rpm]: https://github.com/kubernetes/release/tree/e10a44f8f9a9c08441260574e3d2a8711031fafe/cmd/kubepkg/templates/latest/rpm
+[debbuild]: https://github.com/debbuild/debbuild
+
+#### Integrating OBS with our current release pipeline
+
+As described above, currently, it's up to the Release Manager to create the subproject and packages structure in OBS
+before releasing the first alpha release. This should be the only manual steps required by Release Managers (besides
+the user management, described below). We'll consider automating those steps in the future if possible.
+
+The workflow for publishing packages is:
+
+- Authenticate to OBS via `osc` and pull existing packages from the appropriate **building subproject**
+- Update specs and generate the sources tarball using `kubepkg`
```

Comment:
> Update specs and generate the sources tarball using `kubepkg`

Will this be automated? I think I saw something about building on top of image promoter, but I wasn't sure if that was for this step.

Reply:
> Will this be automated? I think I saw something about building on top of image promoter, but I wasn't sure if that was for this step.

Yes, it would be fully-automated. In fact, we plan to fully-automate everything related to building packages. We'll not build on top of image promoter, but we might come up with a similar pattern (this is not a requirement).

Reply:
I explicitly mentioned that this should be part of `krel`.